src/decoration.cpp

```diff
@@ -380,13 +380,24 @@ void Decoration::redrawPixmap() {
                        inner.width,
                        inner.height - dec->last_actual_rect.height);
     }
-    if (s.title_height() > 0 && s.title_font->data().xFontStruct_) {
-        XSetForeground(g_display, gc, get_client_color(s.title_color));
-        XFontStruct* font = s.title_font->data().xFontStruct_;
-        int font_x_offset = s.padding_left() + s.border_width;
-        XSetFont(g_display, gc, font->fid);
-        XDrawString(g_display, pix, gc, font_x_offset, s.title_height(),
-                    client_->title_().c_str(), client_->title_().size());
+    if (s.title_height() > 0) {
+        FontData& fontData = s.title_font->data();
+        string title = client_->title_();
+        Point2D titlepos = {
+            static_cast<int>(s.padding_left() + s.border_width()),
+            static_cast<int>(s.title_height())
+        };
+        if (fontData.xFontSet_) {
+            XSetForeground(g_display, gc, get_client_color(s.title_color));
```

Comment:
Could be moved before the if because it is used the same way in both branches?

Reply:
It should stay there, because for xft (PR #1091) it is different.

Reply:
Alright! Thanks for clearing that up